src/components/NewsList.tsx

```diff
@@ -97,26 +115,37 @@ export default function NewsList({ searchNews }: NewsListProps) {
     }
   }, [searchNews])
 
+  const isNewsBookmarked = (news: NewsType) => bookmarkedNews.filter( bookmarkedNews => bookmarkedNews.title === news.title).length > 0
+  
+  useEffect(() => {
+    setNewsComponents(newsList.map( news =>
```

Comment:
i think this is not needed. you can directly use newsList to render the components?

Reply:
fixed